### ISSUE TITLE
Question: IO architecture in kata-containers.

### ISSUE THREAD

**Issue by vrfn** [kata-containers/community#177]
Hi, I hope this is the right place to ask a question. If it is not, please let me know.

I have been looking to into kata-containers, and have a question about the way kata handles IO in particular. By default, kata seems to use 9pfs to share the OCI bundle ('docker image') between the hypervisor host and guest. From what I can tell, the only time you would need to share data between the host and guest if a user runs `docker cp`. Although useful in some cases, it is not that frequently used. 

My question is then, why is kata-containers implemented using 9pfs, hampering all IO performance within the container, for such a minor use case? Couldn't kata-containers make use of something along the lines of NFS, in which all local writes within the hypervisor guest container are fast, and just the copying from the guest to the host is slow? 

Thanks!

**Comment by vrfn** [comment]
I just noticed this particular sentence on the virtio-fs [website ](https://virtio-fs.gitlab.io/): 
"Unlike existing approaches, it is designed to offer local file system semantics and performance. " 

This is exactly what I suggested in my initial post above: IO requests within the container should be as fast as local requests on a more ordinary filesystem, which is not the case with 9p. However, from my own tests using kata-containers with virtio-fs enabled, I am not close to getting native speeds. I am also aware that the IO performance issues are being looked into in issues like [this one](https://github.com/kata-containers/runtime/issues/2138).

I guess my question becomes simpler then: is it reasonable to expect near native IO performance once the virtio-fs implementation is more mature? Faster than existing network file systems such as NFS?
